### PR TITLE
feat(app-tools): SSR mode support async entry

### PR DESCRIPTION
**Pull request by 2heal1**
## Summary

support `source.enableAsyncEntry` works on ssr mode

## Details

`source.enableAsyncEntry` will enable async entry , and it works smoothly on csr mode .

But ssr mode doesn't support it ,cause it need some change with entry code and server consumer .

## Checklist

- [x] I have added changeset via `pnpm run change`.
- [x] I have updated the documentation.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: .changeset/chilly-ducks-perform.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+'@modern-js/app-tools': patch
+'@modern-js/prod-server': patch
+---
+
+chore: ssr mode support async entry
```
Comment (chenjiahan) [inline .changeset/chilly-ducks-perform.md:6]: chore -> feat

File: packages/server/prod-server/src/libs/render/ssr.ts [hunk 2]
```diff
@@ -78,7 +78,8 @@ export const render = async (
   context.metrics = createMetrics(context, ctx.metrics);
 
   runner.extendSSRContext(context);
-  const serverRender = require(bundleJS)[SERVER_RENDER_FUNCTION_NAME];
+  const bundleJSContent = await require(bundleJS);
```
Comment (chenjiahan) [inline packages/server/prod-server/src/libs/render/ssr.ts:81]: Is `await require` a valid usage? How about using `await import`?

File: packages/solutions/app-tools/src/analyze/generateCode.ts [hunk 3]
```diff
@@ -285,9 +285,31 @@ export const generateCode = async (
...
+          entryName,
+          config.server.ssr,
+          config.server.ssrByEntries,
+          packageName,
+        );
+        if (ssr) {
+          rawAsyncEntryCode = `
+          export const serverRender = async (...args) => {
+            let exports = await ${rawAsyncEntryCode};
+            if(exports.default instanceof Promise){
```
Comment (chenjiahan) [inline packages/solutions/app-tools/src/analyze/generateCode.ts:299]:
```suggestion
            if (exports.default instanceof Promise){
```

File: .changeset/chilly-ducks-perform.md (hunk 1 above)
Comment (2heal1) [reply .changeset/chilly-ducks-perform.md]: done

File: packages/solutions/app-tools/src/analyze/generateCode.ts [hunk 4]
```diff
@@ -285,9 +285,31 @@ export const generateCode = async (
...
+        let rawAsyncEntryCode = `import('./${ENTRY_BOOTSTRAP_FILE_NAME}');`;
+        const ssr = getEntryOptions(
+          entryName,
+          config.server.ssr,
+          config.server.ssrByEntries,
+          packageName,
+        );
+        if (ssr) {
+          rawAsyncEntryCode = `
+          export const serverRender = async (...args) => {
```
Comment (chenjiahan) [inline packages/solutions/app-tools/src/analyze/generateCode.ts:297]: serverRender -> SERVER_RENDER_FUNCTION_NAME

File: packages/solutions/app-tools/src/analyze/generateCode.ts [hunk 5]
```diff
@@ -285,9 +285,31 @@ export const generateCode = async (
...
+            let exports = await ${rawAsyncEntryCode};
```
Comment (chenjiahan) [inline packages/solutions/app-tools/src/analyze/generateCode.ts:298]: `exports` is a special object that is used in Node.js modules, maybe we can use other variable names.

**Pull request merged**